client/a8c-for-agencies/sections/marketplace/hosting-overview/index.tsx

```diff
@@ -18,20 +19,33 @@ export default function Hosting() {
 	const translate = useTranslate();
 
 	const { selectedCartItems, onRemoveCartItem } = useShoppingCart();
+	const isNarrowView = useBreakpoint( '<660px' );
 
 	return (
 		<Layout
 			className={ classNames( 'hosting-overview' ) }
 			title={ translate( 'Hosting Marketplace' ) }
 			wide
 			withBorder
-			sidebarNavigation={ <MobileSidebarNavigation /> }
 		>
 			<LayoutTop>
-				<LayoutHeader>
-					<Title>{ translate( 'Marketplace' ) }</Title>
+				{ ! isNarrowView ? (
+					<LayoutHeader>
+						<Title>{ translate( 'Marketplace' ) }</Title>
 
-					<Actions>
+						<Actions>
+							<ShoppingCart
+								items={ selectedCartItems }
+								onRemoveItem={ onRemoveCartItem }
+								onCheckout={ () => {
+									page( A4A_MARKETPLACE_CHECKOUT_LINK );
+								} }
+							/>
+						</Actions>
+					</LayoutHeader>
+				) : (
+					<Actions className="a4a-layout-header-mobile">
+						<MobileSidebarNavigation />
```

Comment:
I was thinking that instead of modifying the template according to the viewport, we could use CSS to hide certain elements such as the breadcrumb and title that are not required in the mobile view. Regarding the `MobileSidebarNavigation`, we could place it within the LayoutHeader and ensure that it is visible only on mobile view. We could also remove the `sidebarNavigation` props as they are no longer necessary.

I think this would eliminate the duplication of codes. wdyt?

Reply:
Thanks! This is a good point. Tried to implement things this way, please let me know if you see anything I missed, as it's quite a lot of changes.